controller/src/main/java/cern/jarrace/controller/rest/controller/AgentContainerController.java

```diff
@@ -0,0 +1,104 @@
+ */
+@RestController
+@RequestMapping("/jarrace")
+public class AgentContainerController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AgentContainerController.class);
+    private static final File DEPLOYMENT_DIR = new File(System.getProperty("java.io.tmpdir"));
+
+    final Map<String, List<Entrypoint>> entrypoints = new HashMap<>();
+    final Map<String, String> paths = new HashMap<>();
+
+    @RequestMapping(value = "/container/deploy/{name}", method = RequestMethod.POST)
+    public void deploy(@PathVariable("name") String name, @RequestBody byte[] jar) throws IOException {
+        System.out.println("Deployed " + name);
+        String path = writeFile(name, jar);
+        paths.put(name, path);
+        entrypoints.put(name, new ArrayList<>());
+        List<String> args = new ArrayList<>();
+        args.add("cern.jarrace.agent.AgentContainer");
+        args.add(name);
+        args.add("localhost:8080");
+        startContainer(path, args);
+    }
+
+    @RequestMapping(value = "/service/register/{name}", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE)
+    public void registerService(@PathVariable(value = "name") String name, @RequestBody Entrypoint entrypoint){
+
+        if(entrypoint.getAgentName() == null || entrypoint.getAgentName().isEmpty()) {
+            throw new IllegalArgumentException("Agent name must be different than null and not empty");
+        }
+        if(entrypoint.getPath() == null || entrypoint.getPath().isEmpty()) {
+            throw new IllegalArgumentException("Path must be different than null and not empty");
+        }
+        entrypoints.get(name).add(entrypoint);
+        LOGGER.info(entrypoints.toString());
+    }
+
+    @RequestMapping(value = "/container/list", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public Set<String> listContainers() {
+        return entrypoints.keySet();
+    }
+
+    @RequestMapping(value = "/{name}/entrypoint/list", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public List<Entrypoint> listServices(@PathVariable(value = "name") String containerName) {
+        List<Entrypoint> toReturn = this.entrypoints.get(containerName);
+        if (toReturn == null) {
+            toReturn = new ArrayList<>();
+        }
+        return toReturn;
+    }
+
+    @RequestMapping(value = "/{name}/{entrypoint}/start", method = RequestMethod.GET)
+    public String runService(@PathVariable("name") String name, @PathVariable int entrypoint) throws IOException {
+        List<Entrypoint> entries = entrypoints.get(name);
+        Entrypoint entryPoint = entries.get(entrypoint);
+        List<String> args = new ArrayList<>();
+        args.add("cern.jarrace.agent.AgentRunner");
+        args.add(entryPoint.getAgentName());
+        args.add(entryPoint.getPath());
+        startContainer(paths.get(name), args);
+        return null;
```

Comment:
at the time it was only writing to the stdout of the server but the comuncation channel is still to be decided which is why we decided to do it this way, removed for now